truss/templates/control/control/helpers/truss_patch/model_container_patch_applier.py

```diff
@@ -127,6 +142,48 @@ def _apply_system_package_patch(self, system_package_patch: SystemPackagePatch):
         else:
             raise ValueError(f"Unknown python requirement patch action {action}")
 
+    def _apply_config_patch(self, config_patch: ConfigPatch):
+        self._app_logger.debug(f"Applying config patch {config_patch.to_dict()}")
+        TrussConfig.from_dict(config_patch.config).write_to_yaml_file(
+            Path(self._inference_server_home / config_patch.path)
+        )
+
+    def _apply_env_var_patch(self, env_var_patch: EnvVarPatch):
+        self._app_logger.debug(
+            f"Applying environment variable patch {env_var_patch.to_dict()}"
+        )
+        action = env_var_patch.action
+        ((env_var_name, env_var_value),) = env_var_patch.item.items()
+
+        if action == Action.REMOVE:
+            os.environ.pop(env_var_name, None)
+        elif action in [Action.ADD, Action.UPDATE]:
+            os.environ.update({env_var_name: env_var_value})
+        else:
+            raise ValueError(f"Unknown patch action {action}")
+
+    def _apply_external_data_patch(self, external_data_patch: ExternalDataPatch):
+        self._app_logger.debug(
+            f"Applying external data patch {external_data_patch.to_dict()}"
+        )
+        action = external_data_patch.action
+        if action == Action.REMOVE:
+            item = ExternalDataItem.from_dict(external_data_patch.item)
+            filepath = self._data_dir / item.local_data_path
+            if not filepath.exists():
+                self._app_logger.warning(
+                    f"Could not delete file {filepath}: not found."
+                )
+            else:
+                self._app_logger.debug(f"Deleting file {filepath}")
+                filepath.unlink()
+        elif action == Action.ADD:
```

Comment:
this is specified in [`_calc_external_data_patches`](https://github.com/basetenlabs/truss/pull/377/files#diff-d9ecf761b1a9c01688d6bb78dd8d2d509f6562d1194514d063869d736df66bfaR274) where any change to the `ExternalDataItem` objects will result in a `REMOVE` and `ADD` (I missed the patch sorting altogether for all the patches this PR adds support for will add changes to this branch) but as far as I can tell the ordering should not matter for `REMOVE` + `ADD` since we are modifying these objects based on their `dict` representations, which should always be different if something is changed